src/handler/http/request/pipelines/history.rs

```diff
@@ -150,22 +159,98 @@ pub async fn get_pipeline_history(
     }
 
-    // Helper function to escape pipeline names for SQL LIKE patterns
-    fn escape_like(input: &str) -> String {
-        let mut escaped = String::with_capacity(input.len());
-        for c in input.chars() {
-            match c {
-                '\\' => escaped.push_str(r"\\"),
-                '%' => escaped.push_str(r"\%"),
-                '_' => escaped.push_str(r"\_"),
-                '\'' => escaped.push_str("''"),
-                _ => escaped.push(c),
+    // RBAC: Check permissions and filter by accessible pipelines
+    #[cfg(feature = "enterprise")]
+    let mut permitted_pipeline_names = pipeline_names;
+    #[cfg(not(feature = "enterprise"))]
+    let permitted_pipeline_names = pipeline_names;
+
+    // RBAC: Check permissions for the requested pipeline(s)
+    #[cfg(feature = "enterprise")]
+    let permitted_pipeline_names = {
+        let user_id = &user_email.user_id;
+
+        // If RBAC is enabled, check permissions
+        if get_openfga_config().enabled {
+            let user = match crate::service::users::get_user(Some(&org_id), user_id).await {
+                Some(user) => user,
+                None => {
+                    return MetaHttpResponse::forbidden("User not found");
+                }
+            };
+
+            let role = user.role.to_string();
+
+            // If specific pipeline_name is requested, check access to it
+            if let Some(ref pipeline_name) = query.pipeline_name {
+                let pipeline_obj = format!(
+                    "{}:{}",
+                    OFGA_MODELS.get("pipelines").unwrap().key,
+                    pipeline_name
+                );
+
+                let has_permission = o2_openfga::authorizer::authz::is_allowed(
+                    &org_id,
+                    user_id,
+                    "GET",
+                    &pipeline_obj,
+                    "",
+                    &role,
+                )
+                .await;
+
+                if !has_permission {
+                    return MetaHttpResponse::forbidden(format!(
+                        "Access denied to pipeline '{pipeline_name}'"
+                    ));
+                }
+
+                pipeline_names
+            } else {
+                // List all history - filter by accessible pipelines
+                let mut accessible_pipelines = Vec::new();
+
+                for pipeline_name in &pipeline_names {
+                    let pipeline_obj = format!(
+                        "{}:{}",
+                        OFGA_MODELS.get("pipelines").unwrap().key,
+                        pipeline_name
+                    );
+
+                    let has_permission = o2_openfga::authorizer::authz::is_allowed(
+                        &org_id,
+                        user_id,
+                        "GET",
+                        &pipeline_obj,
+                        "",
+                        &role,
+                    )
+                    .await;
+
```

Comment:
You can use the old logic here, where you get all the pipelines the user has access to